src/link/Wasm.zig

```diff
@@ -1259,6 +1281,36 @@ fn allocateAtoms(self: *Wasm) !void {
     }
 }
 
+fn sortDataSegments(self: *Wasm) !void {
```

Comment:
This might not be very useful to you, but this is what we do in MachO using optional indexes to every known output section that may be in the file https://github.com/ziglang/zig/blob/960c14206094484cee056b8806b9b8f75a84891d/src/link/MachO.zig#L5504 Time-complexity wise, not sure if this results in anything more optimal or whether we should even worry about this given that there are so few indexes to sort.

Reply:
I also use this trick in zld/ELF https://github.com/kubkon/zld/blob/f0de089a7db6dce6b9a14d6645ab43287728c0ca/src/Elf.zig#L1132

Reply:
That would work if all possible segments were known. However, when compiling to a relocatable object file, each data symbol gets its own segment. E.g. `.rodata.my_symbol`. Only for final binaries do we merge them into a single segment. (wasm-ld even allows you to not merge segments). For this reason I choose this option, as I actually remember checking the ELF and MachO implementation when I did this for Coff :)

Reply:
Hmm, so you are saying that in Wasm we permit for a priori "unknown" output sections?

Reply:
Segments, yes. They are not standardized in any way. Only the sections are (types, imports, code, data, etc). Those are already ordered correctly simply within `flush` by emitting them manually in the correct order.